File: dotnet/xmscommon/framing/ContentHeaderBodyFactory.cs
using System;
using jpmorgan.mina.common;
using log4net;

namespace OpenAMQ.Framing
{
    public class ContentHeaderBodyFactory : IBodyFactory
    {
        private static readonly ILog _log = LogManager.GetLogger(typeof(ContentHeaderBodyFactory));

        private static readonly ContentHeaderBodyFactory _instance = new ContentHeaderBodyFactory();

        public static ContentHeaderBodyFactory GetInstance()
        {
            return _instance;
        }

        private ContentHeaderBodyFactory()
        {
            _log.Debug("Creating content header body factory");
        }

        #region IBodyFactory Members

        public IBody CreateBody(ByteBuffer inbuf)
        {
            // all content headers are the same - it is only the properties that differ.
            // the content header body further delegates construction of properties
            return new ContentHeaderBody();
        }

        #endregion
    }
}

File: clients/java/org/openamq/framing/ContentHeaderBodyFactory.java
package org.openamq.framing;

import org.apache.log4j.Logger;
import org.apache.mina.common.ByteBuffer;

/**
 * @author Robert Greig (robert.j.greig@jpmorgan.com)
 */
public class ContentHeaderBodyFactory implements BodyFactory
{
    private static final Logger _log = Logger.getLogger(AMQMethodBodyFactory.class);

    private static final ContentHeaderBodyFactory _instance = new ContentHeaderBodyFactory();

    public static ContentHeaderBodyFactory getInstance()
    {
        return _instance;
    }

    private ContentHeaderBodyFactory()
    {
        _log.debug("Creating content header body factory");
    }

    public AMQBody createBody(ByteBuffer in) throws AMQFrameDecodingException
    {
        // all content headers are the same - it is only the properties that differ.
        // the content header body further delegates construction of properties
        return new ContentHeaderBody();
    }


}
